### ISSUE TITLE
PostgreSQL generated code doesn't maintain precision / scale on NUMERIC or TIME, TIMETZ, TIMESTAMP, TIMESTAMPTZ types

### ISSUE THREAD

**Issue by U1F984** [jOOQ/jOOQ#13919]
### Expected behavior

The generated classes will reflect the precision correctly.

### Actual behavior

The generated classes generate the columns with precision 0, leading to truncation:

```java
public final TableField<FooRecord, LocalDateTime[]> BAR_COLUMN = createField(DSL.name("bar_column"), SQLDataType.LOCALDATETIME(0).getArrayDataType(), this, "");
```

### Steps to reproduce the problem

SQL table:

```sql
CREATE TABLE foo
(
    bar_column   timestamp(3) WITHOUT TIME ZONE[] NOT NULL
);
```

Gradle codegen config:

```kt
generator.apply {
    database.apply {
        name = "org.jooq.meta.postgres.PostgresDatabase"
        inputSchema = "test"
        includes = ".*"
        isOutputSchemaToDefault = true
        isIncludeSystemIndexes = true
        isIncludeSystemSequences = true
    }
}
```

### jOOQ Version

jOOQ Open Source 3.17.2

### Database product and version

PostgreSQL 14.1, compiled by Visual C++ build 1914, 64-bit

### Java Version

OpenJDK 64-Bit Server VM Corretto-17.0.4.8.1 (build 17.0.4+8-LTS, mixed mode, sharing)

### OS Version

Microsoft Windows [Version 10.0.19044.1889]

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.3.3

**Comment by lukaseder** [comment]
Thanks for your report. I can reproduce this. Will look into it right away.

**Comment by lukaseder** [comment]
Hmm, it seems we're not querying this right. The query that is being run to fetch all columns is this one:

```sql
select
  information_schema.columns.column_name,
  information_schema.columns.ordinal_position,
  case
    when (
      information_schema.columns.interval_type like '%YEAR%'
      or information_schema.columns.interval_type like '%MONTH%'
    ) then 'INTERVAL YEAR TO MONTH'
    when (
      information_schema.columns.interval_type like '%DAY%'
      or information_schema.columns.interval_type like '%HOUR%'
      or information_schema.columns.interval_type like '%MINUTE%'
      or information_schema.columns.interval_type like '%SECOND%'
    ) then 'INTERVAL DAY TO SECOND'
    when (
      information_schema.columns.data_type = 'USER-DEFINED'
      and information_schema.columns.udt_name = 'geometry'
    ) then 'geometry'
    else information_schema.columns.data_type
  end as data_type,
  coalesce(
    information_schema.columns.character_maximum_length,
    case
      when information_schema.columns.udt_name in (
        '_varchar', '_bpchar', '_char'
      ) then (pg_catalog.pg_attribute.atttypmod - 4)
    end
  ) as character_maximum_length,
  coalesce(information_schema.columns.datetime_precision, information_schema.columns.numeric_precision) as numeric_precision,
  information_schema.columns.numeric_scale,
  case
    when (
      information_schema.columns.is_identity = 'YES'
      or (
        count(*) filter (where information_schema.columns.is_identity = 'YES') over () = 0
        and lower(information_schema.columns.column_default) ilike 'nextval(''%_seq''::regclass)'
      )
    ) then 'YES'
  end as is_identity,
  information_schema.columns.is_nullable,
  information_schema.columns.generation_expression as generation_expression,
  pg_catalog.pg_attribute.attgenerated as attgenerated,
  case
    when (
      information_schema.columns.is_identity = 'YES'
      or (
        count(*) filter (where information_schema.columns.is_identity = 'YES') over () = 0
        and lower(information_schema.columns.column_default) ilike 'nextval(''%_seq''::regclass)'
      )
    ) then null
    else information_schema.columns.column_default
  end as column_default,
  coalesce(information_schema.columns.domain_schema, information_schema.columns.udt_schema) as udt_schema,
  coalesce(information_schema.columns.domain_name, information_schema.columns.udt_name) as udt_name,
  pg_catalog.pg_description.description
from information_schema.columns
  join pg_catalog.pg_namespace
    on information_schema.columns.table_schema = pg_catalog.pg_namespace.nspname
  join pg_catalog.pg_class
    on (
      pg_catalog.pg_class.relname = information_schema.columns.table_name
      and pg_catalog.pg_class.relnamespace = pg_catalog.pg_namespace.oid
    )
  join pg_catalog.pg_attribute
    on (
      pg_catalog.pg_attribute.attrelid = pg_catalog.pg_class.oid
      and pg_catalog.pg_attribute.attname = information_schema.columns.column_name
    )
  left outer join pg_catalog.pg_description
    on (
      pg_catalog.pg_description.objoid = pg_catalog.pg_class.oid
      and pg_catalog.pg_description.objsubid = information_schema.columns.ordinal_position
    )
where (
  information_schema.columns.table_schema = 'public'
  and information_schema.columns.table_name = 't_dates'
)
order by information_schema.columns.ordinal_position;
```

It produces the type `ARRAY` without precision / scale information, and the `udt_name` is just `_timestamp`.

**Comment by lukaseder** [comment]
Ah, I see. There had been a fix for `CHAR(n)[]` and `VARCHAR(n)[]`: https://github.com/jOOQ/jOOQ/issues/11658, but not for numeric / timestamp arrays, which still use ordinary `COLUMNS.NUMERIC_PRECISION`, `COLUMNS.NUMERIC_SCALE`, `COLUMNS.DATETIME_PRECISION`

**Comment by lukaseder** [comment]
I'll fix this for `NUMERIC[]`, `NUMERIC(p)[]` and `NUMERIC(p,s)[]` as well, since those precisions/scales aren't being fetched correctly, either.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.4 (https://github.com/jOOQ/jOOQ/issues/13921), 3.16.10 (https://github.com/jOOQ/jOOQ/issues/13922)

**Comment by lukaseder** [comment]
The fix applies also to CockroachDB and YugabyteDB. It doesn't affect H2, HSQLDB, whose implementations were already correct.

**Comment by ChristopherZellermann** [comment]
I have this issue as well but with jOOQ 3.14.16 and Java 8 + Maven.

Will it be possible to apply a fix to the 3.14 branch ? Should I open a new bug for this ?

Christopher

**Comment by lukaseder** [comment]
It seems this fix was still incorrect? https://github.com/jOOQ/jOOQ/issues/14736
